src/System.Windows.Forms/src/System/Windows/Forms/Control.cs

```diff
@@ -2099,8 +2099,72 @@ public virtual Font Font
             [param: MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(ActiveXFontMarshaler))]
             set
             {
-                SetFont(value);
+                var local = (Font)Properties.GetObject(s_fontProperty);
+                bool localChanged = false;
+                if (value is null)
+                {
+                    if (local is not null)
+                    {
+                        localChanged = true;
+                    }
+                }
+                else
+                {
+                    localChanged = local is null ? true : !value.Equals(local);
+                }
+
+                if (localChanged)
+                {
+                    // If Font bing assigned is equal to the current default Font, we just set the property bag no need to raise OnFontChangedEvent.
```

Comment:
I suggest to invert if to a) reduce nesting and b) make this comment closer to the case it applies to.
```suggestion
                    // If the value being assigned is the same as the current font, we do not need to raise FontChangedEvent.
```

If the value is the same as the current font - why are we updating the property bag?

Reply:
If Font is default property bag will be empty but if user is setting same default value on the Form, we need to set it to prevent scaling from the parent Font changed.